# Ans 1

In [1]:
print(1.5e200 *2.0e210)


inf


inf = infinity

In [2]:
print(3.4e-320 / 1e100)

0.0


# Ans 2

In [3]:
print(1/3)

0.3333333333333333


In [4]:
print(format(1/3, '.3f'))

0.333


In [5]:
print(format(1/3, '.1f'))

0.3


In [6]:
print(6*1/3)

2.0


In [7]:
print(1/3 + 1/3 + 1/3 + 1/3 + 1/3 + 1/3)

1.9999999999999998


# Ans 3

Difference between a and b is :
In a) value after decimal is printed till maximum value 
In b) we have asked to print till 3 decimal points using format specifier

Difference between d and e is:
In d) as per associativity rule 6 will be multiplied 1 and then the resultant number will be divided by 3
But in e) as / and + has different associativity BODMAS rule is applied

# Ans 4

In [8]:
print(format(22/7, '.5f'))

3.14286


In [16]:
print("%-10s" % ('Hello', ))

Hello     


In [17]:
print(format("Hello", "<10"))

Hello     


In [19]:
print(format("Hello", ">10"))

     Hello


In [47]:
print(format("Hello", "#^15"))

#####Hello#####


# Ans 5

In [48]:
print(ord('a'))
print(ord('A'))
print(ord('z'))
print(ord('Z'))

97
65
122
90


In [42]:
print(ord('0'))
print(ord('9'))
print(ord(' '))
print(ord('\n'))
print(ord('\t'))

48
57
32
10
9


In [45]:
print(chr(35))
print(chr(67))
print(chr(50))
print(chr(99))

#
C
2
c


For lower case : 97 - 122
For upper case : 65 - 90
For digits : 48 -57

# Ans 6

In [49]:
print("abc","\n","bcd")

abc 
 bcd


In [51]:
print("abc\nbcd")

abc
bcd


In [52]:
print(1,'\n',2)

1 
 2


In [53]:
print("hello \n\n!!")

hello 

!!


In [54]:
print("This is '\t' means tab")

This is '	' means tab


4 spaces are printed using \t

In [55]:
print("This","is","different",sep='\t')

This	is	different


In [56]:
print("This","is","different",sep='#')

This#is#different


In [57]:
print("This","is","different",sep='\n')

This
is
different


In [58]:
print("This","is","different",end='\n')

This is different


In [60]:
print(10 / 2, 10 // 2.0, 101 / 3, 101 // 3, 101 % 3, 2 ** 4, 3 ** 2, sep="\n")

5.0
5.0
33.666666666666664
33
2
16
9


In [61]:
abs(-10) # give absolute value

10

In [62]:
divmod(101,3) # divmod gives two values , interger division and remainder

(33, 2)

In [63]:
print(8 << 1) # 8 x 2^1

16


In [64]:
print(8 << 2)

32


In [65]:
print(32 >> 2) # 32 / 2^2

8


In [66]:
print(16 >> 1)

8


# Ans 8

In [ ]:
a,b,c = int(input("Enter marks : ").split())
total = int(a) + int(b) + int(c)
print(total)

Q9. Manually calculate output of following expressions using precedence and associativity rules 
and then verify output using python shell
a. 2 + 3 * (4 -1)
b. 2 + ((3 *4) - 8)
c. (2 + 3) * 4
d. -10 + 25 / (16 + 12)
e. 2 ** 2 ** 2
f. (3 ** 2) ** 3
g. x = y = 3 + 5
h. 8/2*(2+2)

In [25]:
print(2 + 3 * (4 -1))

11


In [12]:
print(2 + ((3 *4) - 8))

6


In [13]:
print((2 + 3) * 4)

20


In [14]:
print(-10 + 25 / (16 + 12))

-9.107142857142858


In [16]:
print(2 ** 2 ** 2)

256


In [17]:
print((3 ** 2) ** 3)

729


In [18]:
print(x = y = 3 + 5)

SyntaxError: invalid syntax (2559457631.py, line 1)

In [23]:
x = y = 3 + 5
print(x)
print(y)


8
8


In [24]:
print(8/2*(2+2))

16.0


Q10. Check output of following 
a. x = y = 10
b. x1 = 20 = y1

In [26]:
x = y = 10
print(x)
print(y)

10
10


In [27]:
x1 = 20 = y1

SyntaxError: cannot assign to literal (1868644801.py, line 1)

Q11. What is output of following
a. x=10
y=34.4
print(x and y)
print(x or y)

In [31]:
x=10
y=34.4
print(x and y)  #if x and y are true then print y else 0
print(x or y)   

34.4
10


# IF ELSE

1.	A student will not be allowed to sit in exam if his/her attendence is less than 75%.  
Take following input from user Number of classes held Number of classes attended. And print percentage of class attended  
Is student is allowed to sit in exam or not. 

In [9]:
import math
class_held=int(input("Enter number of classes held : "))
class_att=int(input("Enter number of classes attended : "))
percentage=(class_att/class_held)*100
print("Attendence percentage is : ",math.ceil(percentage), '%') #use math.ceil() to print value of nearest integer
if percentage >= 75:
    print("You are allowed to sit in the exam !!")
else:
    print("You are not allowed to sit in the exam :( ")


Enter number of classes held : 100
Enter number of classes attended : 76
Attendence percentage is :  76 %
You are allowed to sit in the exam !!


2. accept amount from user and find the minimum number notes required to get the amount amount =512  
Notes: 2000,500,100,50,10,5,2,1  

500-1 note  
10  - 1 note  
2-  1 coin  

amount=20550  
2000 – 10 note  
500 – 1 note  
50 -1 note

In [3]:
amount=int(input("Enter your amount: "))

if (amount>=2000):
    note2k=amount//2000
    print("2000 notes: ", note2k)
    rem=amount%2000
    amount=rem
if (2000>amount>=500):
    note500=amount//500
    print("500 notes: ", note500)
    rem=amount%500
    amount=rem
if (500>amount>=200):
    note200=amount//200
    print("200 notes: ", note200)
    rem=amount%200
    amount=rem
if (200>amount>=100):
    note100=amount//100
    print("100 notes: ", note100)
    rem=amount%100
    amount=rem

if (100>amount>=20):
    note20=amount//20
    print("20 notes: ", note20)
    rem=amount%20
    amount=rem
if (20>amount>=10):
    note10=amount//10
    print("10 notes: ", note10)
    rem=amount%10
    amount=rem
if (10>amount>=5):
    note5=amount//5
    print("5 notes: ", note5)
    rem=amount%5
    amount=rem
if (5>amount>=2):
    note2=amount//2
    print("2 notes: ", note2)
    rem=amount%2
    amount=rem
if (2>amount>=1):
    note1=amount//1
    print("1 notes: ", note1)
    rem=amount%1
    amount=rem



Enter your amount: 5668
2000 notes:  2
500 notes:  3
100 notes:  1
20 notes:  3
5 notes:  1
2 notes:  1
1 notes:  1


In [22]:
amount=int(input("Enter your amount: "))
arr=[2000,500,100,50,20,10,5,2,1]
for i in arr:
    if (amount>=i):
        note=amount//i
        print("{0} notes: ".format(i), note)
        rem=amount%i
        amount=rem
#    print(i)

Enter your amount: 15423
2000 notes:  7
500 notes:  2
100 notes:  4
20 notes:  1
2 notes:  1
1 notes:  1


3.	Modify the above question Q1 to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [7]:
import math
class_held=int(input("Enter number of classes held : "))
class_att=int(input("Enter number of classes attended : "))
med=input("he/she has medical cause or not? ( 'Y' or 'N' )")
percentage=(class_att/class_held)*100
print("Attendence percentage is : ",math.ceil(percentage), '%') #use math.ceil() to print value of nearest integer
if percentage >= 75:
    print("You are allowed to sit in the exam !!")
elif (med == "Y"):
    print("You are allowed to sit in the exam !!")
else:
    print("You are not allowed to sit in the exam :( ")

Enter number of classes held : 100
Enter number of classes attended : 65
he/she has medical cause or not? ( 'Y' or 'N' )Y
Attendence percentage is :  65 %
You are allowed to sit in the exam !!


4.	A school has following rules for grading system:  
a.	Below 25 - F  
b.	25 to 45 - E  
c.	45 to 50 - D  
d.	50 to 60 - C  
e.	60 to 80 - B  
f.	Above 80 - A  
Ask user to enter marks and print the corresponding grade.

In [25]:
marks=int(input("Enter your Marks: "))
if marks<25:
    print("F")
elif 25<=marks<45:
    print("E")
elif 45<=marks<50:
    print("D")
elif 50<=marks<60:
    print("C")
elif 60<=marks<80:
    print("B")
elif 80<=marks<=100:
    print("A")
else:
    print("Please enter valid input")


Enter your Marks: 56
C


In [32]:
marks=int(input("Enter your Marks: "))
arr = [0, 25, 45, 50, 60, 80, 100]
arr_ch = ['F', 'E', 'D', 'C', 'B', 'A']
for i in range(6):
    if arr[i]<=marks<arr[i+1]:
        print(arr_ch[i])

Enter your Marks: 46
D


5.	  If  x = 2 y = 5 z = 0  then find values of the following expressions and Print the output of following statements  
a.	x == 2  
b.	x != 5  
c.	x != 5 & y >= 5  
d.	z != 0 | x == 2  
e.	not(y < 10)  

In [35]:
x = 2
y = 5 
z = 0
print(x == 2)
print(x != 5)
print(x != 5 & y >= 5)
print(z != 0 | x == 2)
print(not(y < 10))

True
True
True
True
False


6.	Accept number from user and check whether it is divisible by 5 and 11 if divisible then display appropriate message. 

In [41]:
num=int(input("Enter a number: "))
if (num%5==0) & (num%11==0):
    print("Divisible by both 5 and 11")
else:
    print("Not divisible")

Enter a number: 14
Not divisible


7.	Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :  
             Unit                                    Price    
First 100 units                                       no charge  
Next 100 units                                        Rs 5 per unit  
After 200 units                                       Rs 10 per unit  
(For example if input unit is 350 than total bill amount is Rs2000)  

In [44]:
unit = int(input("Enter the units : "))
charge = 0
if unit>200:
    charge = charge + (unit-200) * 10
    unit = 200
if 200>=unit>100:
    charge = charge + (unit-100) * 5
    unit = 100
if 100>unit:
    charge = charge + (unit-0) * 0
    unit = 0
print(charge)

Enter the units : 350
2000


8.	Write a program to check whether the last digit of a number( entered by user ) is divisible by 3 or not.  


In [46]:
num=int(input("Enter a number: "))
digit = num % 10
if digit%3==0:
    print("Number is divisible by 3")
else:
    print("Number is not divisible by 3")

Enter a number: 452
Number is not divisible by 3


9.	Write a program to check whether an years is leap year or not the year is leap if it satisfies following condition  
•	It the year is divisible by 100 o If it is divisible by 100, then it should also be divisible by 400 then it is leap year  
•	otherwise, all other years divisible by 4 and not divisible by 100 then it is leap year.  

In [59]:
year=int(input("Enter a year: "))
if year % 100 == 0:
    if year % 400 == 0:
        print("Leap year")
    else:
        print("Not a leap year")
elif year % 4 == 0:
    print("Leap year")
else:
    print("Not a leap year")

Enter a year: 2023
Not a leap year


10. Write a program to accept the price of a bike and display the road tax and insurance to be paid according to the following criteria . also display total amount to be paid.  
      
        Cost price (in Rs)           Tax                Inssurance  
        > 100000                     15 %                   20%  
        > 50000 and <= 100000        10%                    8%  
        <= 50000                     5%                     5%   

6.	To display the cube of the number upto given an integer. If the given integer is 5, then display cube of 1 to 4.

In [3]:
num = int(input("Enter a number : "))
for i in range (1,num):
    print(i*i*i)

Enter a number : 5
1
8
27
64


7. Accept 20 numbers from user and display sum of only even numbers

In [8]:
total=0
for i in range (1,20):
    a = int(input("Enter number: "))
    if (a%2==0):
        total+=a
print(total)

Enter number: 1
Enter number: 2
Enter number: 3
Enter number: 4
Enter number: 5
Enter number: 6
Enter number: 7
Enter number: 8
Enter number: 9
Enter number: 0
Enter number: 1
Enter number: 2
Enter number: 3
Enter number: 4
Enter number: 5
Enter number: 6
Enter number: 7
Enter number: 8
Enter number: 9
40


8. Ask user number of terms to be generated of a series. 
generate numbers for the following series and find its addition 
[9 + 99 + 999 + 9999+……..] 

In [11]:
num=int(input("Enter number of terms: "))
total=0
print("Terms are: ", end=" ")
for i in range(1,num+1):
    a= '9'*i
    if i < (num):
        print(a, '+ ', end="")
        total+= int(a)
    else:
        print(a)
        total+= int(a)
print("Total is :", total)

Enter number of terms: 2
Terms are:  9 + 99
Total is : 108


Take a number from user and print odd numbers 

In [2]:
num=int(input("Enter a number: "))

for i in range(1,num+1,-2):
    print(i)

Enter a number: -15
1
-1
-3
-5
-7
-9
-11
-13


9.	Write a program in python to display the sum of the series [ 1+x+x^2/2!+x^3/3!+....]. Go to the editor 
Test Data : 
Input the value of x :3 Input number of terms : 5 Expected Output : 
The sum is : 16.375000 


In [6]:
import math
x=int(input("Enter value of x : "))
num=int(input("Enter numer of terms : "))
total=0
for i in range(num):
    total+= ((pow(x, i))/math.factorial(i))
print(total)
    

Enter value of x : 4
Enter numer of terms : 1
1.0


10.	Write a program in python to find the sum of the series [ x - x^3 + x^5 –x^7+x^9-x^11+ ......]. Go to the editor 
Test Data : 
Input the value of x :2 Input number of terms : 5 Expected Output : 
The values of the series: 
2 
-8 
32 
-128 
512 
The sum = 410

In [9]:
import math
x=int(input("Enter value of x : "))
num=int(input("Enter numer of terms : "))
total=0
term=1
for i in range(1,num*2,2):
    if term % 2 !=0:
        total+= (pow(x, i))
    else:
        total-= (pow(x, i)) 
    term+=1
print(total)
    

Enter value of x : 2
Enter numer of terms : 5
410
